File: languages/go/entity/identity/issuer_state_receipt.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IssuerStateReceipt represents a receipt for the issuer's state.
type IssuerStateReceipt struct {
	TxHash string
}

func NewIssuerStateReceiptFromProto(s *proto.IssuerStateReceipt) IssuerStateReceipt {
	if s == nil {
		return IssuerStateReceipt{}
	}
	return IssuerStateReceipt{
		TxHash: s.TxHash,
	}
}

func (i IssuerStateReceipt) ToProto() *proto.IssuerStateReceipt {
	return &proto.IssuerStateReceipt{
		TxHash: i.TxHash,
	}
}

File: languages/js/src/entity/identity/issuer_state_receipt.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";

/**
 * Represents a receipt for the issuer's state.
 */
export class IssuerStateReceipt {
  txHash: string;

  /**
   * Constructs an IssuerStateReceipt object with the specified parameters.
   * @param txHash
   */
  constructor(txHash: string) {
    this.txHash = txHash;
  }

  public toProto(): identityEntitiesProto.IssuerStateReceipt {
    return identityEntitiesProto.IssuerStateReceipt.fromPartial({
      txHash: this.txHash
    });
  }

  static fromProto(
    r: identityEntitiesProto.IssuerStateReceipt
  ): IssuerStateReceipt {
    return new IssuerStateReceipt(r.txHash);
  }
}
